pkg/controller/collection/collection_controller.go

```diff
@@ -24,14 +25,15 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller"
 	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 var log = logf.Log.WithName("controller_collection")
+var cIDRegex = regexp.MustCompile("^[a-z]+[a-z0-9-]*[a-z0-9]+$")
```

Comment:
With this regex, a collection ID of `a` will fail.  I'm not sure how to write a regex that will catch this case... maybe if the length of the ID is 1, we can just check to see if it's a valid lower case letter?

Reply:
Thanks for catching this Tim. Updated a new commit.